bindings/go/src/ucx/worker.go

```diff
@@ -261,8 +261,7 @@ func (w *UcpWorker) NewListener(listenerParams *UcpListenerParams) (*UcpListener
 // received on this worker.
 func (w *UcpWorker) SetAmRecvHandler(id uint, flags UcpAmCbFlags, cb UcpAmRecvCallback) error {
 	var amHandlerParams C.ucp_am_handler_param_t
-	cbId := register(cb)
-	idToWorker[cbId] = w
+	cbId := register(&UcpAmRecvCallbackBundle{cb: cb, worker: w})
```

Comment:
Please check whether we need to change the signature of `func register(cb UcpCallback) uint64`

Reply:
`type UcpCallback interface{}` so it can be assigned by any type AFAIK.
It compiled and Milvus run successfully with it.

1. We can change the type to be more specific, we are now storing bundle not callback so it's more correct.
2. We can use golang sync map and get rid of this register function, but it's not typesafe.
3. @Artemy-Mellanox suggested using [golang Handles](https://pkg.go.dev/runtime/cgo#Handle), With it we can get rid of this map and save additional map lookup, but it's require newer Golang version then what UCX specify/use.

Reply:
As discussed f2f with @ofirfarjun7 , we'll go with (3) above (handles), it's supported starting with Golang v1.18 and allows cleaner code.